lib/features/rules_filter_creator/presentation/rules_filter_creator_controller.dart

```diff
@@ -206,90 +214,72 @@ class RulesFilterCreatorController extends BaseMailboxController {
+          listEmailRuleFilterActionSelected.add(RuleFilterActionArguments.newAction(action));
+        }
+
+        if (_currentTMailRule!.action.markAsSeen == true) {
+          EmailRuleFilterAction? action = EmailRuleFilterAction.maskAsSeen;
+          listEmailRuleFilterActionSelected.add(RuleFilterActionArguments.newAction(action));
+        }
+
+        if (listEmailRuleFilterActionSelected.length >= maxCountAction) {
+          isShowAddAction.value = false;
+        } else {
+          isShowAddAction.value = true;
+        }
+
+        _newRuleName = _currentTMailRule!.name;
+        _setValueInputField(inputRuleNameController, _newRuleName ?? '');
         break;
     }
     inputRuleNameFocusNode.requestFocus();
   }
 
+  MailboxId? getSpamMailboxId() {
+    return findMailboxNodeByRole(PresentationMailbox.roleJunk)?.item.id
+      ?? findMailboxNodeByRole(PresentationMailbox.roleSpam)?.item.id;
+  }
+
   void _setValueInputField(TextEditingController? controller, String value) {
     controller?.value = controller.value.copyWith(
         text: value,
         selection: TextSelection.collapsed(offset: value.length));
   }
 
-  void _setUpMailboxSelected() {
-    if (_currentTMailRule != null) {
-      final mailboxIdsOfRule = _currentTMailRule!.action.appendIn.mailboxIds;
-      final spamMailboxId = findMailboxNodeByRole(PresentationMailbox.roleJunk)?.item.id
-        ?? findMailboxNodeByRole(PresentationMailbox.roleSpam)?.item.id;
-      for (var mailboxId in mailboxIdsOfRule) {
-        if (mailboxId != spamMailboxId) {
-          final mailboxNode = findMailboxNodeById(mailboxId);
-          if (mailboxNode != null) {
-            mailboxSelected.value = mailboxNode.item;
-          }
-        }
-      }
-      RuleFilterActionArguments newRuleFilterAction = MoveMessageActionArguments(mailbox: mailboxSelected.value);
-      for (var filterAction in listEmailRuleFilterActionSelected) {
-        if (filterAction is MoveMessageActionArguments) {
-          listEmailRuleFilterActionSelected[listEmailRuleFilterActionSelected.indexOf(filterAction)] = newRuleFilterAction;
+  void _setUpRuleFilterActions() {
+    if (_currentTMailRule!.action.appendIn.mailboxIds.isNotEmpty == true) {
+      final mailboxNode = findMailboxNodeById(_currentTMailRule!.action.appendIn.mailboxIds.first);
+      if (mailboxNode != null) {
+        if (mailboxNode.item.isSpam) {
+          listEmailRuleFilterActionSelected.add(MarkAsSpamActionArguments());
+        } else {
+          mailboxSelected.value = mailboxNode.item;
+          listEmailRuleFilterActionSelected.add(
+            MoveMessageActionArguments(mailbox: mailboxNode.item)
+          );
         }
+      } else {
+        mailboxSelected.value = PresentationMailbox.unifiedMailbox;
+        listEmailRuleFilterActionSelected.add(
+          MoveMessageActionArguments(mailbox: PresentationMailbox.unifiedMailbox)
+        );
       }
-      listEmailRuleFilterActionSelected.refresh();
     }
```

Comment:
```suggestion
    if (_currentTMailRule!.action.appendIn.mailboxIds.isNotEmpty != true) return;
    
    final mailboxNode = findMailboxNodeById(_currentTMailRule!.action.appendIn.mailboxIds.first);
    if (mailboxNode == null) {
      mailboxSelected.value = PresentationMailbox.unifiedMailbox;
      listEmailRuleFilterActionSelected.add(
        MoveMessageActionArguments(mailbox: PresentationMailbox.unifiedMailbox)
      );
    } else if (mailboxNode.item.isSpam) {
      listEmailRuleFilterActionSelected.add(MarkAsSpamActionArguments());
    } else {
      mailboxSelected.value = mailboxNode.item;
      listEmailRuleFilterActionSelected.add(
        MoveMessageActionArguments(mailbox: mailboxNode.item)
      );
    }
```

Avoid nesting `if` statements

Reply:
Your suggestion will make us lose the processing behind it. There should be no return call here. We can separate the function to handle it.